apps/ledger-live-desktop/src/renderer/modals/Send/fields/RecipientFieldNamingService.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { memo, useEffect, useMemo } from "react";
+import { Transaction, TransactionStatus } from "@ledgerhq/live-common/generated/types";
+import { Account, AccountBridge } from "@ledgerhq/types-live";
+import { TFunction } from "react-i18next";
+
+import { useNamingService } from "@ledgerhq/live-common/naming-service/index";
+import RecipientFieldBase from "./RecipientFieldBase";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+
+type Props = {
+  account: Account;
+  transaction: Transaction;
+  autoFocus?: boolean;
+  status: TransactionStatus;
+  onChangeTransaction: (tx: Transaction) => void;
+  t: TFunction;
+  label?: React.ReactNode;
+  initValue?: string;
+  resetInitValue?: () => void;
+  value: string | string;
+  bridge: AccountBridge<Transaction>;
+  onChange: (recipient: string, maybeExtra?: Record<string, CryptoCurrency>) => Promise<void>;
+};
+
+const RecipientField = ({
+  t,
+  account,
+  transaction,
+  onChangeTransaction,
+  autoFocus,
+  status,
+  label,
+  value = "",
+  bridge,
+  onChange,
+}: Props) => {
+  const namingServiceResponse = useNamingService(value);
+  const hasValidatedName = useMemo(() => namingServiceResponse.status === "loaded", [
+    namingServiceResponse.status,
+  ]);
+
+  useEffect(() => {
+    if (
+      hasValidatedName &&
+      (transaction.recipient === transaction.recipientName || value !== transaction.recipientName)
+    ) {
+      onChangeTransaction(
+        bridge.updateTransaction(transaction, {
+          recipient: (namingServiceResponse as { address: string }).address,
```

Comment:
Don't know if you want to fix that, but I had a look about type narrowing in those cases and the solution would be to change:
```typescript
 const hasValidatedName = useMemo(() => namingServiceResponse.status === "loaded", [
    namingServiceResponse.status,
  ]);
```

into:
```typescript
 const hasValidatedName = useCallback((nsResponse: UseNamingServiceResponse): nsResponse is { status: "loaded"; address: string } => nsResponse.status === "loaded", []);
```

and use `hasValidatedName(namingServiceResponse)` in your condition instead. Might be worth to have that hook in LLC and create a specific type `UseNamingServiceResponseLoaded` ?

Might a bit of work for just avoiding a cast, so it's up to you 👍

Reply:
Nice thank you, will try to update it if it's help readability

Reply:
Not so trivial still working on it :(